cpp/src/arrow/filesystem/path_util.cc

```diff
@@ -139,6 +140,13 @@ util::string_view RemoveLeadingSlash(util::string_view key) {
   return key;
 }
 
+Status AssertNoTrailingSlash(const std::string& key) {
+  if (key.back() == '/') {
+    return NotAFile(key);
```

Comment:
I would take `util::string_view` instead but `NotAFile` only accepts `const std::string&`. Would it be alright if I moved "arrow/filesystem/util_internal.h" to use `util::string_view`?

Reply:
Definitely ok!